### ISSUE TITLE
Nixpkgs repo is getting unreasonably large

### ISSUE THREAD

**Issue by copumpkin** [copumpkin/nixpkgs#78]
Doing a fresh clone now pulls down almost 200MB. I haven't tried to see where all the space is going, but it doesn't seem like a good situation.

**Comment by jwiegley** [comment]
I wonder if in the prehistory there were large binaries included in the repository. A typical git clone has a `.git` directory that is smaller than its working tree, whereas that's not anywhere near the case with `nixpkgs`.

The problem is, the only way to rectify a problem like this is to rewrite history and get rid of those dead contents.  It may be easier to just start over with another repo.

**Comment by copumpkin** [comment]
There were binaries checked in at one point, yeah. I investigated the large repo when I first became involved with the project and it turned out there were some obsolete branches holding onto some unnecessary stuff and we freed up a bunch of space. But now it's grown significantly again in the past few months.

It's not hard to rewrite the history to get rid of the large binaries. There's even a dedicated program for it, but we'd need to coordinate the community to do it, since it's pretty disruptive to people's workflows. Long-lived branches and so on are going to be pretty painful to merge. I don't know what the best course of action is, but this seems bad.
